phoenix-core/src/main/java/org/apache/phoenix/compile/ProjectionCompiler.java

```diff
@@ -641,19 +702,23 @@ public <T> T accept(ExpressionVisitor<T> visitor) {
             return null;
         }
     }
-    private static void serailizeArrayIndexInformationAndSetInScan(StatementContext context, List<Expression> arrayKVFuncs,
-            List<KeyValueColumnExpression> arrayKVRefs) {
+
+    private static void serializeServerParsedExpressionInformationAndSetInScan(
+            StatementContext context, String serverParsedExpressionAttribute,
+            List<Expression> serverParsedKVFuncs,
+            List<KeyValueColumnExpression> serverParsedKVRefs) {
         ByteArrayOutputStream stream = new ByteArrayOutputStream();
         try {
             DataOutputStream output = new DataOutputStream(stream);
-            // Write the arrayKVRef size followed by the keyvalues that needs to be of type arrayindex function
-            WritableUtils.writeVInt(output, arrayKVRefs.size());
-            for (Expression expression : arrayKVRefs) {
+            // Write the arrayKVRef size followed by the keyvalues that needs to be of
+            // type arrayindex+json function
+            WritableUtils.writeVInt(output, serverParsedKVRefs.size());
+            for (Expression expression : serverParsedKVRefs) {
                     expression.write(output);
             }
-            // then write the number of arrayindex functions followeed by the expression itself
-            WritableUtils.writeVInt(output, arrayKVFuncs.size());
-            for (Expression expression : arrayKVFuncs) {
+            // then write the number of arrayindex+json functions followed by the expression itself
```

Comment:
Again `+` here seems to imply that everything is serialized together

Reply:
sure. will update it.